components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.sms/src/main/java/org/wso2/carbon/event/output/adapter/sms/SMSEventAdapter.java

```diff
@@ -0,0 +1,120 @@
+import org.wso2.carbon.event.output.adapter.sms.internal.ds.SMSEventAdapterServiceValueHolder;
+import org.wso2.carbon.event.output.adapter.sms.internal.util.SMSEventAdapterConstants;
+
+import java.util.Map;
+
+public final class SMSEventAdapter implements OutputEventAdapter {
+
+    private static final Log log = LogFactory.getLog(SMSEventAdapter.class);
+    private OutputEventAdapterConfiguration eventAdapterConfiguration;
+    private Map<String, String> globalProperties;
+
+    private ServiceClient serviceClient;
+
+    public SMSEventAdapter(OutputEventAdapterConfiguration eventAdapterConfiguration, Map<String, String> globalProperties) {
+        this.eventAdapterConfiguration = eventAdapterConfiguration;
+        this.globalProperties = globalProperties;
+    }
+
+    @Override
+    public void init() throws OutputEventAdapterException {
+        String SMSNo = eventAdapterConfiguration.getStaticProperties().get(SMSEventAdapterConstants.ADAPTER_MESSAGE_SMS_NO);
+        ConfigurationContext configContext = SMSEventAdapterServiceValueHolder.getConfigurationContextService().getClientConfigContext();
+        try{
+            if (configContext != null) {
+                serviceClient = new ServiceClient(configContext, null);
+            } else {
+                serviceClient = new ServiceClient();
+            }
+            Options options = new Options();
+            options.setProperty(Constants.Configuration.ENABLE_REST, Constants.VALUE_TRUE);
+            options.setTo(new EndpointReference("sms://" + SMSNo));
+            serviceClient.setOptions(options);
+        } catch (AxisFault axisFault){
+            String systemType = serviceClient.getAxisConfiguration().getTransportOut("sms").getParameter("systemType").toString();
+            String systemID = serviceClient.getAxisConfiguration().getTransportOut("sms").getParameter("systemId").toString();
+            String password = serviceClient.getAxisConfiguration().getTransportOut("sms").getParameter("password").toString();
+            String host = serviceClient.getAxisConfiguration().getTransportOut("sms").getParameter("host").toString();
+            String port = serviceClient.getAxisConfiguration().getTransportOut("sms").getParameter("port").toString();
+            String phoneNumber = serviceClient.getAxisConfiguration().getTransportOut("sms").getParameter("phoneNumber").toString();
+            String msg = "Error in creating client, " +
+                    "with configuration parameters systemType: " + systemType + ", systemID: " + systemID + ", password: " + password
+                    + ", host: " + host + ", port: " + port + ", phoneNumber: " + phoneNumber
+                    + ", to: " + eventAdapterConfiguration.getStaticProperties().get(SMSEventAdapterConstants.ADAPTER_MESSAGE_SMS_NO) + ".";
+            throw new OutputEventAdapterException(msg, axisFault);
+        }
```

Comment:
We cannot have SMS No as static property.. I have changed the implementation and commited to repo because BPS team had some concers.. Please check and raise if you have any concern on this..